hls4ml/model/optimizer/passes/bn_quant.py

```diff
@@ -63,6 +40,26 @@ def config_cpp(self):
 
         return self._config_template.format(**params)
 
+    def set_thresholds(self, scale, bias, ternary_threshold=0.5):
```

Comment:
Moved the setting of the thresholds from the layer `initialize` to a separate function because the `BatchNormalization` layer from which the thresholds are derived may have been inserted by another optimization pass (`QKerasFactorizeAlpha` or new `ExtractTernaryThreshold`). In that case the layer doesn't exist in the original model and , e.g., `variance = self.model.get_weights_data(original_name, 'moving_variance')` won't work